spec/integration/user_index_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe 'Test Index Page', type: :feature do
+  describe 'GET index' do
+    before(:each) do
+      @first_user = User.create(name: 'Mohammed', photo: 'image1.png', bio: 'bio1', posts_counter: 1)
+      @first_user.save!
+      @second_user = User.create(name: 'Ahmed', photo: 'image2.png', bio: 'bio2', posts_counter: 3)
+      @second_user.save!
+      @third_user = User.create(name: 'Marwan', photo: 'image3.png', bio: 'bio3', posts_counter: 5)
+      @third_user.save!
+    end
+
+    it 'shows the users username' do
+      visit root_path
+      expect(page).to have_content('Mohammed')
+      expect(page).to have_content('Ahmed')
+      expect(page).to have_content('Marwan')
+    end
+
```

Comment:
- [ ] **Required:** You are required to have a test for a link to the show page. Kindly include it.